### PR TITLE
Updated dev:refresh to allow for BYODB.

### LINKED ISSUE #172: Add support for custom database dump to `dev:refresh` command
## Description

Developers sometimes need to work off of a non-production database dump. One such case is when a long-running migration branch is in active development and there are other development efforts that branch off of that branch. Devs branching off of the migration branch often find themselves needing to import a database dump from a “staging” or “dev” environment that has been set up to track the migration branch.

Currently, developers who want to use `dev:refresh` might do something like:
1. Run `composer robo dev:refresh ddev`. _This imports the prod database._
2. Run `ddev import-db --file=./some-non-prod-db-dump.sql.gz`.

That certainly gets the job done, but on projects with database dump sizes in the gigabytes, it takes a very long time to actually refresh one’s dev environment. It would be really useful to be able to pass an option to the `dev:refresh` command which, when present, makes Robo skip the prod database download altogether and use the provided dump instead.

## Issue Owner

- @agarzola 

## Acceptance Criteria

- [ ] A new option is available for the `dev:refresh` command that accepts a local path to a database dump file.
    - e.g. `--db=./path-to-dump.sql.gz`
- [ ] When this option is used, `dev:refresh` skips the database download step and imports the provided database dump instead.

## Alternatives Considered

n/a

## Additional Context

Requested in our `#devops` channel.

### PR DESCRIPTION
**Pull request by markdorison**
## Description
Updated `dev:refresh` to allow a user to BYODB (Bring Your Own Database).

## Motivation / Context
Closes #172

## Testing Instructions / How This Has Been Tested
Tested locally with two different sites/scenarios.

## Screenshots
CleanShot 2023-10-19 at 14 03 33@2x

### CODE REVIEW

**Comment by ghost** [comment]
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=173&src=https%3A%2F%2Fgithub.com%2FChromaticHQ%2Fusher">Click on the image</a> for a new way to code review</b>
</summary>

####

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=173&src=https%3A%2F%2Fgithub.com%2FChromaticHQ%2Fusher)

#### Legend
CodeSee Map legend
</details>

File: src/Robo/Plugin/Commands/DevelopmentModeBaseCommands.php [hunk 1]
```diff
@@ -67,12 +67,20 @@ public function __construct()
...
+     * @option db
+     *   Provide a path to a database dump to be used instead of downloading the latest dump.
      */
-    public function databaseRefreshDdev(string $siteName = 'default'): Result
+    public function databaseRefreshDdev(string $siteName = 'default', array $options = ['db' => '']): Result
     {
         $this->io()->title('DDEV database refresh.');
 
-        $dbPath = $this->databaseDownload($siteName);
+        ['db' => $dbPath] = $options;
```
Comment (adamzimmermann) [inline src/Robo/Plugin/Commands/DevelopmentModeBaseCommands.php:77]: 🎩

**Review by adamzimmermann — APPROVED** [review]
The logic checks out. 👏🏼  

Nice and simple. 💯

**Review by agarzola — APPROVED** [review]
Exquisite!

**Pull request merged**